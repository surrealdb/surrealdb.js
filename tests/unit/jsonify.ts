import { assertSnapshot } from "https://deno.land/std@0.224.0/testing/snapshot.ts";
import {
	Decimal,
	Duration,
	GeometryCollection,
	GeometryLine,
	GeometryMultiPolygon,
	GeometryPoint,
	GeometryPolygon,
	jsonify,
	RecordId,
	StringRecordId,
	Table,
	UUID,
} from "../../mod.ts";

Deno.test("jsonify matches snapshot", async function (t) {
	const json = jsonify(
		{
			rid: new RecordId("some:thing", "under_score"),
			str_rid: new StringRecordId("⟨some:thing⟩:under_score"),
			dec: new Decimal("3.333333"),
			dur: new Duration("1d2h"),
			geo: new GeometryCollection([
				new GeometryPoint([1, 2]),
				new GeometryMultiPolygon([
					new GeometryPolygon([
						new GeometryLine([
							new GeometryPoint([1, 2]),
							new GeometryPoint([3, 4]),
						]),
						new GeometryLine([
							new GeometryPoint([5, 6]),
							new GeometryPoint([7, 8]),
						]),
					]),
				]),
				new GeometryPolygon([
					new GeometryLine([
						new GeometryPoint([1, 2]),
						new GeometryPoint([3, 4]),
					]),
					new GeometryLine([
						new GeometryPoint([5, 6]),
						new GeometryPoint([7, 8]),
					]),
				]),
			]),

			tb: new Table("some super _ cool table"),
			uuid: UUID.parse("92b84bde-39c8-4b4b-92f7-626096d6c4d9"),
			date: new Date("2024-05-06T17:44:57.085Z"),
			undef: undefined,
			null: null,
			num: 123,
			float: 123.456,
			true: true,
			false: false,
			string: "I am a string",
		},
	);

	await assertSnapshot(t, json);
});
