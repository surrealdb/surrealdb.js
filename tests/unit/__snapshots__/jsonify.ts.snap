export const snapshot = {};

snapshot[`jsonify matches snapshot 1`] = `
{
  date: "2024-05-06T17:44:57.085Z",
  dec: "3.333333",
  dur: "26h",
  false: false,
  float: 123.456,
  geo: {
    geometries: [
      {
        coordinates: [
          "1",
          "2",
        ],
        type: "Point",
      },
      {
        coordinates: [
          [
            [
              [
                "1",
                "2",
              ],
              [
                "3",
                "4",
              ],
            ],
            [
              [
                "5",
                "6",
              ],
              [
                "7",
                "8",
              ],
            ],
          ],
        ],
        type: "MultiPolygon",
      },
      {
        coordinates: [
          [
            [
              "1",
              "2",
            ],
            [
              "3",
              "4",
            ],
          ],
          [
            [
              "5",
              "6",
            ],
            [
              "7",
              "8",
            ],
          ],
        ],
        type: "Polygon",
      },
    ],
    type: "GeometryCollection",
  },
  null: null,
  num: 123,
  rid: "⟨some:thing⟩:under_score",
  str_rid: "⟨some:thing⟩:under_score",
  string: "I am a string",
  tb: "some super _ cool table",
  true: true,
  uuid: "92b84bde-39c8-4b4b-92f7-626096d6c4d9",
}
`;
